PhysicsTools/PyTorch/test/testTorchSimpleDnnCUDA.cc

```diff
@@ -0,0 +1,74 @@
+#include <torch/script.h>
+#include "testBaseCUDA.h"
+#include <iostream>
+#include <memory>
+#include <vector>
+
+class testSimpleDNNCUDA : public testBasePyTorchCUDA {
+  CPPUNIT_TEST_SUITE(testSimpleDNNCUDA);
+  CPPUNIT_TEST(test);
+  CPPUNIT_TEST_SUITE_END();
+
+public:
+  void test() override;
+};
+
+CPPUNIT_TEST_SUITE_REGISTRATION(testSimpleDNNCUDA);
+
+void testSimpleDNNCUDA::test() {
+  if (!cms::cudatest::testDevices())
+    return;
+
+  std::vector<edm::ParameterSet> psets;
+  edm::ServiceToken serviceToken = edm::ServiceRegistry::createSet(psets);
+  edm::ServiceRegistry::Operate operate(serviceToken);
+
+  // Setup the CUDA Service
+  edmplugin::PluginManager::configure(edmplugin::standard::config());
+
+  std::string const config = R"_(import FWCore.ParameterSet.Config as cms
+process = cms.Process('Test')
+process.add_(cms.Service('ResourceInformationService'))
+process.add_(cms.Service('CUDAService'))
+)_";
+  std::unique_ptr<edm::ParameterSet> params;
+  edm::makeParameterSets(config, params);
+  edm::ServiceToken tempToken(edm::ServiceRegistry::createServicesFromConfig(std::move(params)));
+  edm::ServiceRegistry::Operate operate2(tempToken);
+  edm::Service<CUDAInterface> cuda;
+  std::cout << "CUDA service enabled: " << cuda->enabled() << std::endl;
```

Comment:
Setting up the `CUDAService` seems unnecessary for now, but maybe it becomes useful in the near future?

Reply:
I replicated the structure used in the TensorFlow tests: I think this was to make sure that GPUs are available and used for the CUDA tests

Reply:
Reminding myself, `PhysicsTools/TensorFlow` uses `ResourceInformationService`
https://github.com/cms-sw/cmssw/blob/3244058986397a861e53d069633555e3754d0bd3/PhysicsTools/TensorFlow/src/TensorFlow.cc#L36
and for that to be filled correctly the `CUDAService` is needed there.

If you foresee similar structure here, I'm fine with keeping this (presently unnecessary) Service complexity in this test.

Reply:
Indeed I foresee to add a similar structure in the PyTorch interface to correctly setup the backend and check for the accelerator presence.